wagtail/admin/views/pages.py

```diff
@@ -517,7 +517,7 @@ def edit(request, page_id):
 
 
 def delete(request, page_id):
-    page = get_object_or_404(Page, id=page_id)
+    page = get_object_or_404(Page, id=page_id).specific
```

Comment:
The `.specific` triggers a second SQL query. To avoid that, we can pass directly a queryset to `get_object_or_404` that will prefetch the specific page:
```python
page = get_object_or_404(Page.objects.specific(), id=page_id)
```

Reply:
@BertrandBordage I don't think this will make any difference to the number of queries - the `specific()` method on a queryset will execute as many additional queries as there are distinct page types in the resultset (i.e. one extra query, in this case).

Reply:
Ah, you’re right, sorry about that.